### ISSUE TITLE
Subworkflows always run single-threaded

### ISSUE THREAD

**Issue by abmorris** [snakemake/snakemake#208]
**Snakemake version**
5.10.0 and 5.9.1

Bug not present in 5.8.2 or earlier, tried with 5.9.0 but it crashed with `ModuleNotFoundError: No module named 'snakemake.deployment'`

**Describe the bug**
When using `subworkflow` and the `-j` (or `--cores` or `--jobs`) command line option, the subworkflow is executed as if `-j` is not specified at all

```
Provided cores: 1 (use --cores to define parallelism)
```

**Logs**

```
$ snakemake -j 2
Building DAG of jobs...
Executing subworkflow sub.
Building DAG of jobs...
Using shell: /bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	2	write_file
	2

[Mon Jan 27 17:34:58 2020]
rule write_file:
    output: 0.txt
    jobid: 0
    wildcards: n=0

[Mon Jan 27 17:34:58 2020]
Finished job 0.
1 of 2 steps (50%) done

[Mon Jan 27 17:34:58 2020]
rule write_file:
    output: 1.txt
    jobid: 1
    wildcards: n=1

[Mon Jan 27 17:34:58 2020]
Finished job 1.
2 of 2 steps (100%) done
Complete log: /lhcb5/users/admorris/scratch/test_snake/sub/.snakemake/log/2020-01-27T173458.770831.snakemake.log
Executing main workflow.
Using shell: /bin/bash
Provided cores: 2
Rules claiming more threads will be scaled down.
Job counts:
	count	jobs
	1	merge_files
	1

[Mon Jan 27 17:34:58 2020]
rule merge_files:
    input: /lhcb5/users/admorris/scratch/test_snake/sub/0.txt, /lhcb5/users/admorris/scratch/test_snake/sub/1.txt
    output: numbers.txt
    jobid: 0

[Mon Jan 27 17:34:58 2020]
Finished job 0.
1 of 1 steps (100%) done
Complete log: /lhcb5/users/admorris/scratch/test_snake/sub/.snakemake/log/2020-01-27T173458.770831.snakemake.log
```

**Minimal example**
`Snakefile`:
```
subworkflow sub:
	configfile:
		"config.yml" # issue #24
	workdir:
		"sub"

rule merge_files:
	input:
		[sub(f"{i}.txt") for i in range(2)]
	output:
		"numbers.txt"
	shell:
		"cat {input} > {output}"
```
`sub/Snakefile`:
```
rule write_file:
	output:
		"{n}.txt"
	shell:
		"echo {wildcards.n} > {output}"
```
`config.yml`:
```
dummy: "whatever" # issue #24
```

**Comment by lumpiluk** [comment]
Sadly, this has been marked as "wontfix" [almost 5 years ago](https://bitbucket.org/snakemake/snakemake/issues/119/subworkflows-parallel):

> Snakemake provides two mechanisms for modularization: 1. the include instruction 2. the subworkflow instruction
>
> The first was the initial method. It will lead to a single big DAG, behaving fully parallel. It works very well. You can see good examples for this in the Snakemake workflow archive.
>
> The second was created for one special purpose: To use files from another, already existing workflow, that should be kept untouched. Here, I really wanted maximum separation.
>
> Personally, I think all modularization can be done with include. It has one drawback, which is missing namespace support. But in the filesystem, where the results of your rules will reside you don't have implicit namespaces either. You need to solve that by using folders or prefixes. The same can be done with functions, variables and rules within the Snakefile.
>
> So, for maximum parallelism, use include, for maximum separation, use subworkflow.

In my current situation, using prefixes would seem very tedious. I also see no straight-forward way using includes to keep paths pointing to the same location whether `snakemake` is run in the root directory or in `sub/` (#206 appears to be related).

**Comment by abmorris** [comment]
But it worked fine up to and including 5.8.2? Indeed the relative path stuff really complicates using includes. It could be worked around by deducing absolute paths, but that's not particularly elegant and potentially problematic.

**Comment by engenegr** [comment]
I'm trying to figure out how to make this forking with minimal code changes but appears it is too complex task for unknown library.

**Comment by pdagosto** [comment]
I have this issue also. I have a subworkflow that uses pipes between steps (required for efficiency) and a main workflow that cannot use pipes (because of the nature of the programs being run - and no, I cannot modify them to do so). 

The subworkflow executes fine on its own with --cores 13. However, when included as a subworkflow to the main workflow it fails to execute because it is only allocated 1 core even though --cores 13 was specified on the command line.

The idea that "include" is sufficient for all modular development is badly mistaken in my opinion because 1) the lack of name spacing and 2) the need in some cases to isolate individual parts of a complex workflow so that each part can depend only on the output of another part without being exposed to all of its internal behavior. The latter aspect is also extremely helpful in making the code more understandable and allowing individual testing of each part.

**Comment by mschubi** [comment]
I have the same issue. It really seems to be a bug since it used to work in the version 5.8.2... Subworkflows are really convenient but if they can't work in parallel, it's really annoying.

**Comment by goi42** [comment]
Just to emphasize, I don't see why subworkflows couldn't simply be passed the same `--cores` argument as the main workflow.
